okhttp/src/main/java/io/grpc/transport/okhttp/OkHttpWritableBufferAllocator.java

```diff
@@ -38,8 +38,24 @@
  * The default allocator for {@link OkHttpWritableBuffer}s used by the OkHttp transport.
  */
 class OkHttpWritableBufferAllocator implements WritableBufferAllocator {
+
+  private final int maxDataLength;
+
+  /**
+   * Construct a new instance
+   * @param maxDataLength the maximum allowed length of a DATA frame.
```

Comment:
Hmm... maybe it is better to do this here, as the frame.split() may choose to copy (if we get misaligned between Buffer's Segments and the HTTP/2 streams).